app/js/lib/foxbox/box-link.js

```diff
@@ -0,0 +1,131 @@
+import EventDispatcher from './common/event-dispatcher';
```

Comment:
So then name ```link``` comes from the fact that both ```local``` and ```tunnel``` origins lead to the same ```box```, so we have one ```box``` we're connected to and possibly two ```routes``` or ```links``` to it. Initially I had ```box-route```, but I liked ```box-link``` more - tell me if sounds confusing to you.